tests/unit/moduleapi/datatype.tcl

```diff
@@ -113,4 +113,26 @@ start_server {tags {"modules"}} {
 
         assert_equal 1 [llength $keys]    
     }
+    test {SCAN module datatype with case sensitive} {
+        r flushdb
+        populate 1000
+        r datatype.set foo 111 bar
+        # TODO: remove this in redis 8.0
+        set type "TEST___DT"
```

Comment:
when reading this test, i don't recall the case with which the module registers the type.
i originally thought that we'll search both "TEST___DT" and "test___dt" and match the results.
but maybe using an odd "tEsT___dT" will prove the point and be clearer?